bitnami/thanos/templates/receive/statefulset.yaml

```diff
@@ -71,13 +71,17 @@ spec:
             - --grpc-address=0.0.0.0:10901
             - --http-address=0.0.0.0:10902
             - --remote-write.address=0.0.0.0:19291
-            - --receive.replication-factor=1
+            - --receive.replication-factor={{ .Values.receive.replicationFactor }}
             - --objstore.config=$(OBJSTORE_CONFIG)
             - --tsdb.path=/var/thanos/receive
             - --label={{ .Values.receive.replicaLabel }}="$(NAME)"
             - --label=receive="true"
             - --tsdb.retention=15d
-            - --receive.local-endpoint=127.0.0.1:10901
+            {{- if and (eq .Values.receive.replicaCount 1) .Values.receive.config }}
+            - --receive.local-endpoint=127.0.0.1:{{ .Values.receive.service.grpc.port }}
```

Comment:
Why are you using the service port? It should use the container port (regardless what port you open at service level)

Reply:
I will hardcode it here.